### ISSUE TITLE
admin.accounts.find/update/remove(object)

### ISSUE THREAD

**Issue by courajs** [hoodiehq/hoodie-account-client#96]
[`Find`][find], [`update`][update], and [`remove`][remove] are documented as accepting an object with an `id`, `username`, or `token` property, but currently only work when passed a string id.
When passed an object they mistakenly try to `GET` `http://localhost:4200/hoodie/account/api/accounts/[object%20Object] `

[find]: https://github.com/hoodiehq/hoodie-account-client/tree/master/admin#adminaccountsfind
[update]: https://github.com/hoodiehq/hoodie-account-client/tree/master/admin#adminaccountsupdate
[remove]: https://github.com/hoodiehq/hoodie-account-client/tree/master/admin#adminaccountsremove

**Comment by gr2m** [comment]
Great catch! That would make a great starter issue for https://github.com/hoodiehq/camp/issues :) Anyone feel free to go ahead and create one, otherwise I’ll get back to it as soon as I can

**Comment by courajs** [comment]
I'd enjoy creating my first starter issue 😀

How do we find a user by username? I couldn't figure it out from an (admittedly quick) look at `hoodie-account-server`.

Same for token. What are the tokens, and where do they come from?

We could also keep it super short and only add the `.id` option in the starter issue. Or do just id and username if the token is more complicated or not supported yet

On Sun, May 29, 2016 at 4:35 PM Gregor Martynus

<
mailto:Gregor Martynus <notifications@github.com>
> wrote:

Great catch! That would make a great starter issue for
https://github.com/hoodiehq/camp/issues
:) Anyone feel free to go ahead and create one, otherwise I’ll get back to it as soon as I can

—

You are receiving this because you authored the thread.

Reply to this email directly,
https://github.com/hoodiehq/hoodie-account-client/issues/96#issuecomment-222381264
, or
https://github.com/notifications/unsubscribe/ABg8McOMmuXa3KAXdJzU6h-lgvohTjIdks5qGfiNgaJpZM4IpH8h
.

https://github.com/hoodiehq/hoodie-account-client/issues/96#issuecomment-222381264

**Comment by gr2m** [comment]
Tokens are currently out of scope. They can later be used for one-time logins, e.g. for password resets. 

This one is actually tricky, I get back to it as soon as I can :) If you like we can also work on it together

**Comment by courajs** [comment]
We could in the meantime add `{id: ...}` support, and throw `NotImplementedError` or similar for username/token. Do you think that kind of thing is worth it in general? To match documentation?

**Comment by gr2m** [comment]
yeah, we did that in the past. `NotImplementedError` is great, and then ideally add URL to an open issue to fix it :)
